src/internal_events/aws_ecs_metrics.rs

```diff
@@ -43,53 +23,91 @@ impl InternalEvent for AwsEcsMetricsRequestCompleted {
 #[derive(Debug)]
 pub struct AwsEcsMetricsParseError<'a> {
     pub error: serde_json::Error,
-    pub url: &'a str,
+    pub endpoint: &'a str,
     pub body: Cow<'a, str>,
 }
 
 impl<'a> InternalEvent for AwsEcsMetricsParseError<'_> {
     fn emit_logs(&self) {
-        error!(message = "Parsing error.", url = %self.url, error = %self.error);
+        error!(
+            message = "Parsing error.",
+            endpoint = %self.endpoint,
+            error = ?self.error,
+            stage = "processing",
+            error_type = "parse_failed",
+        );
         debug!(
             message = %format!("Failed to parse response:\\n\\n{}\\n\\n", self.body.escape_debug()),
-            url = %self.url,
+            endpoint = %self.endpoint,
             internal_log_rate_secs = 10
         );
     }
 
     fn emit_metrics(&self) {
         counter!("parse_errors_total", 1);
+        counter!(
+            "component_errors_total", 1,
```

Comment:
Should this include an `error` tag? I've left this comment elsewhere so to explain my perspective -- my reading of the [component spec](https://github.com/vectordotdev/vector/blob/master/docs/specs/component.md#error) makes me feel like it should but curious if it's more discretionary. cc: @bruceg

Reply:
I aggree, the `error` field should be here